File: patterns/java/src/refactoring_guru/patterns/strategy/internet_order/strategies/PayStrategy.java
package refactoring_guru.patterns.strategy.internet_order.strategies;

/**
 * EN: Common interface for all strategies.
 * 
 * RU: Общий интерфейс всех стратегий.
 */
public interface PayStrategy {
    boolean pay(int paymentAmount);
    void collectPaymentDetails();
}

File: patterns/cs/src/RefactoringGuru/Strategy/Example/Strategies/PayStrategy.cs
namespace RefactoringGuru.Strategy.Example.Strategies
{
    /**
     * EN: Common interface for all strategies.
     * 
     * RU: Общий интерфейс всех стратегий.
     */
    interface PayStrategy
    {
        bool Pay(int paymentAmount);
        void CollectPaymentDetails();
    }
}
